### DISCUSSION TITLE
How to pre-load dynamic rules in SvelteKit/Vite?

### DISCUSSION THREAD

**Discussion by alechko** [unocss/unocss#3881] [Q&A]
Hi,

I'm trying to understand how can I pre-load rules like width or colors in SvelteKit/Vite.
I have a component that generates html elements from a store function and whenever I try applying dynamic classes like `bg-gray` or `w-2` the css rule is not being loaded unless I use the class in a svelte component.

for example, when trying to apply conditional background color for elements like this:
```ts
...
  const levelElm = levels.find((v) => v.getAttribute('lvl') === `${o + 1}`);
  if (!levelElm) {
    return;
  }
  let bg;
  if (o >= levelCount) {
    bg = 'transparent';
  } else if (o <= 8 && o > 5) {
    bg = 'white';
  } else if (o <= 5 && o > 3) {
    bg = 'gray-3';
  } else if (o <= 3) {
    bg = 'gray-4';
  }
  levelElm.className = `w2 h1 level bg-${bg}`;
```
the classes `w2`, `h1`, `bg-transparent`, `bg-white` etc. are not being recognized and loaded, only if I create a dummy element in a component and list all the classes, something like:
```
<div class="w2 h1 bg-transparent bg-white bg-gray-3 bg-gray-4"></div>
```
the classes are being loaded.

my unocss.config.ts looks like this:
```ts
import { defineConfig } from 'unocss';
import presetMini from '@unocss/preset-mini';
import extractorSvelte from '@unocss/extractor-svelte';

export default defineConfig({
	extractors: [extractorSvelte()],
	presets: [presetMini()]
});
```

and my vite.config.ts looks like this:
```ts
import { sveltekit } from '@sveltejs/kit/vite';
import { defineConfig } from 'vitest/config';
import UnoCSS from 'unocss/vite';

export default defineConfig({
	plugins: [UnoCSS(), sveltekit()],
	test: {
		include: ['src/**/*.{test,spec}.{js,ts}']
	}
});
```

TIA!

**Comment by schelmo** [comment]
you can add your classes to the `safelist` option:
https://unocss.dev/guide/extracting#safelist

**Reply by alechko** [reply]
thank you, this looks exactly what I'm looking for, much appreciated!
